test/Interop/SwiftToCxx/class/nested-classes-in-cxx-execution.cpp

```diff
@@ -0,0 +1,21 @@
+// RUN: %empty-directory(%t)
+
+// RUN: %target-swift-frontend %S/nested-classes-in-cxx.swift -enable-library-evolution -typecheck -module-name Classes -clang-header-expose-decls=all-public -emit-clang-header-path %t/classes.h
+
+// RUN: %target-interop-build-clangxx -std=c++17 -c %s -I %t -o %t/swift-classes-execution.o
+
+// RUN: %target-interop-build-swift %S/nested-classes-in-cxx.swift -enable-library-evolution -o %t/swift-classes-execution -Xlinker %t/swift-classes-execution.o -module-name Classes -Xfrontend -entry-point-function-name -Xfrontend swiftMain
+
+// RUN: %target-codesign %t/swift-classes-execution
+// RUN: %target-run %t/swift-classes-execution
+
+// REQUIRES: executable_test
+
+#include "classes.h"
+
+int main() {
+  using namespace Classes;
+  auto x = makeRecordConfig();
+  RecordConfig::File::Gate y = x.getGate();
+  RecordConfig::AudioFormat z = x.getFile().getFormat();
+}
```

Comment:
Can you do more in this test here? Some things I can think of:

- Directly instantiate nested classes (i.e., referring to constructors)
- Call some methods and computed properties of nested classes (which you'll need to define)
- Assert some things about the values of `y.prop` (should be equal to `-80`) and `z` (should be `RecordConfig::AudioFormat::ALAC`).

Reply:
Added more tests including asserts. We do not expose Swift initializers as C++ constructors but as `init` methods. Added an example of that.